test/e2e/framework/util.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2016 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"k8s.io/client-go/1.5/kubernetes"
+	"k8s.io/client-go/1.5/pkg/api"
+	"k8s.io/client-go/1.5/pkg/api/v1"
+	"k8s.io/client-go/1.5/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/1.5/pkg/labels"
+	"k8s.io/client-go/1.5/pkg/util/wait"
+	"k8s.io/client-go/1.5/pkg/util/yaml"
+
+	"github.com/coreos/prometheus-operator/pkg/spec"
+)
+
+func CreateDeployment(kclient kubernetes.Interface, ns string, deploy *v1beta1.Deployment) error {
+	if _, err := kclient.Extensions().Deployments(ns).Create(deploy); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func WaitForPodsReady(kclient kubernetes.Interface, ns string, timeout time.Duration, expectedRunning int, opts api.ListOptions) (*v1.PodList, error) {
```

Comment:
Looks like this function and a few others exist in the exact or similar way in the operator code as well (probably both).
It would probably make sense to factor these out into `pkg/k8sutil`